store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreSchemaTest.java

```diff
@@ -70,101 +82,187 @@ public void setUp() throws Exception {
+        expected.addAll(Arrays.asList("a", "b", "c"));
+
+        assertTrue(expected.equals(graphIds));
+    }
+
+    @Test
+    public void shouldGetCorrectDefaultViewForAChosenGraphOperation() throws
+            Exception {
+        // Given
+        addGroupCollisionGraphs();
+
+        // When
+        final CloseableIterable<? extends Element> a = fStore.execute(new OperationChain.Builder()
+                .first(new GetAllElements.Builder()
+                        //No view so makes default view, should get only view compatible with graph "a"
+                        .option(FederatedStoreConstants.KEY_OPERATION_OPTIONS_GRAPH_IDS, "a")
+                        .build())
+                .build(), testContext);
+
+        // Then
+        assertNotNull(a);
+        assertFalse(a.iterator().hasNext());
+    }
+
+    @Test
+    public void shouldBeAbleToGetElementsWithOverlappingSchemas() throws
+            OperationException {
+        // Given
+        addOverlappingPropertiesGraphs();
+
+        fStore.execute(new AddElements.Builder()
+                .input(new Edge.Builder()
+                        .group("e1")
+                        .source("source1")
+                        .dest("dest1")
+                        .property("prop1", "value1")
+                        .build())
+                .build(), testContext);
+
+        fStore.execute(new AddElements.Builder()
+                .input(new Edge.Builder()
+                        .group("e1")
+                        .source("source1")
+                        .dest("dest1")
+                        .property("prop1", "value1")
+                        .property("prop2", "value2")
+                        .build())
+                .build(), testContext);
+
+        // When
+        final CloseableIterable<? extends Element> a = fStore.execute(new GetElements.Builder()
+                .input(new EntitySeed("source1"))
+                .view(new View.Builder()
+                        .edge("e1", new ViewElementDefinition.Builder()
+                                .properties("prop2")
+                                .build())
+                        .build())
+                .build(), testContext);
+
+        assertNotNull(a);
+        final ArrayList<Element> results = new ArrayList<>();
+        final CloseableIterator<? extends Element> aIterator = a.iterator();
+        while (aIterator.hasNext()) {
+            Element result = aIterator.next();
+            results.add(result);
+        }
+
+        // Then
+        assertEquals(results.size(), 2);
+    }
+
+    @Test
+    public void shouldBeAbleToGetSchemaWithOverlappingSchemas() throws
+            OperationException {
+        // Given
+        addOverlappingPropertiesGraphs();
+
+        // When
+        final Schema s = fStore.execute(new GetSchema.Builder()
+                .build(), testContext);
+
+        // Then
+        assertEquals(s.validate(), new ValidationResult());
```

Comment:
I think I'd like it concreted in the test equals object or equals json.

Reply:
`s.validate().isValid` is a boolean, `s.validate()` is a `ValidationResult`

Reply:
```
// Then
        ValidationResult validate = s.validate();
        assertTrue(validate.isValid(), validate.getErrorString());
```